### PR TITLE
Feat: Add Kontak Darurat Page

### LINKED ISSUE #325: Kontak darurat page
## UI Design

image

image

image

## Implementation Model

Using Netlify Collection, we create the collection for the following sections:

- Contacts (list of name, description, image, and external link)
- Vaccine section (lists of text + external link)
- Oxygen section (lists of text + external link)

## Tasks

- [ ] Define Netlify CMS content structure
- [ ] Create Kontak Darurat page following the design in Figma
- [ ] Add Kontak Darurat CTA to homepage

### PR DESCRIPTION
**Pull request by syauqy**
Closes #325 

## Description

I added collections for Kontak Darurat page.
image

#### Contact list
image

#### Vaccine section
image

#### Oxygen section
image

## Current Tasks

- [x] Define Netlify CMS content structure
- [x] Create Kontak Darurat page following the design in Figma
- [x] Add Kontak Darurat CTA to homepage
- [ ] Change kontak image from 'img' to 'next/image' (?)
- [x] Write integration test for `kontak-darurat` page and its components.

### CODE REVIEW

**Comment by resir014** [comment]
We should probably expedite linking Netlify CMS with Cloudinary at #395 before this gets merged, so content writers can easily upload optimised pictures for the cards. Would be helpful with this page + Donasi Mandiri (#329). cc: @ekafyi @zainfathoni

**Comment by zainfathoni** [comment]
Agreed. @ekafyi, how much effort do you think we need to get it implemented?

**Comment by syauqy** [comment]
Here's the kontak darurat page.
Currently, I'm still using `img` instead `next/image`.
I did little change on `StackedLink` component to be reusable for this page.
I added `Accordion` component that wrapped the `StackedLink`.  
kontak darurat page

**Comment by syauqy** [comment]
Kontak darurat CTA added to `homepage-start`.
Added several lines on `homepage-start-test` as well.
image

File: components/home/homepage-emergency-cta.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            {createElement(ContactIcon, {
+              className: "w-6 h-6 text-brand-500",
+            })}
+          </div>
+          <div className="ml-4 space-y-1">
+            <h3 className="text-base leading-5 font-semibold">
+              <Link href="/kontak-darurat">
+                <a
+                  className="helper-link-cover"
+                  rel="nofollow noopener noreferrer"
```
Comment (resir014) [inline components/home/homepage-emergency-cta.tsx:26]:
```suggestion
```

Don't include these if it's an internal link.

**Review by resir014 — CHANGES_REQUESTED** [review]

File: components/home/homepage-emergency-cta.tsx (hunk 1 above)
Comment (syauqy) [reply components/home/homepage-emergency-cta.tsx]: noted, thanks 🙏

**Review by resir014 — CHANGES_REQUESTED** [review]
Please reimplement the accordion with `Disclosure` from [Headless UI](https://headlessui.dev/react/disclosure) - It adds additional a11y features that comply with [WAI-ARIA guidelines](https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html) for accordion.

We should create UI components with accessibility in mind, always.

**Comment by syauqy** [comment]
> Please reimplement the accordion with `Disclosure` from [Headless UI](https://headlessui.dev/react/disclosure) - It adds additional a11y features that comply with [WAI-ARIA guidelines](https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html) for accordion.
> 
> We should create UI components with accessibility in mind, always.

Ahh, sorry. I was going to use headless UI for the accordion. But I'm worried to add a new package to the project. 
I'll change the accordion to disclosure. Thanks.

**Comment by resir014** [comment]
@syauqy Headless UI is already installed. You don't have to worry about it.

File: __tests__/pages/kontak-darurat.test.tsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    render(<VaccineSection vaccine_section={vaccine_section} />);
+    const disclosure_title = screen.getByText(/Mau Vaksin COVID-19?/i);
+    expect(disclosure_title).toBeVisible();
+  });
+
+  it("render the stacked vaccine links correctly", () => {
+    // eslint-disable-next-line no-lone-blocks
+    {
+      // eslint-disable-next-line array-callback-return
+      vaccine_section.map((link, i) => {
```
Comment (zainfathoni) [inline __tests__/pages/kontak-darurat.test.tsx:115]: Why do we need this ESLint disabling rule as well?
Tests are not supposed to return any value, so instead of `map`, you should have used `forEach`. But again, I would avoid doing loop in the tests by introducing the edge cases manually through `@jackfranklin/test-data-bot` builders.

Moreover, if you need to repeat a lot of cases, you can use `test.each()` instead.

File: lib/kontak-darurat/emergency-contacts.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  readonly emergency_contacts: ContactDetail[];
+};
+export type ContactDetail = {
+  readonly name: string;
+  readonly url: string;
+  readonly image: string;
+  readonly description: string;
+};
+
+export default emergencyContacts as unknown as Contact;
```
Comment (zainfathoni) [inline lib/kontak-darurat/emergency-contacts.ts:15]: Please move all the type definition files under the `lib/content` directory.

**Review by zainfathoni — CHANGES_REQUESTED** [review]
Thanks for working on it @syauqy!
However, I think there are some changes needed before we can merge this PR to the `main` branch. 😁 🙏

File: public/admin/config.yml [hunk 4]
```diff
@@ -45,26 +45,10 @@ collections:
...
-        }
-      - {
-          label: "Link",
-          name: "link",
-          widget: "string",
-        }
+      - { label: "Title", name: "title", widget: "string" }
+      - { label: "Date", name: "date", widget: "date" }
+      - { label: "Link Text", name: "link_text", widget: "string" }
+      - { label: "Link", name: "link", widget: "string" }
```
[the remark this answers is not in the document]
Comment (syauqy) [reply public/admin/config.yml:51]: eh? Is that from the `informasi-terbaru` collection? I didn't touch any collection other than `emergency-page`. Sorry it could be autoformatted while I merging the branch. I'll fix it. 🙏

File: components/layout/stacked-link-disclosure.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    <Disclosure>
+      {({ open }) => (
+        <>
+          <Disclosure.Button className="py-2 flex justify-between w-full">
+            <p className="text-gray-800 text-md font-semibold">{title}</p>
+            <span className="text-gray-600">
+              <ChevronDownIcon
+                className={`h-8 w-8 ${
+                  open
+                    ? "transform rotate-180 duration-300 ease"
```
[the remark this answers is not in the document]
Comment (syauqy) [reply components/layout/stacked-link-disclosure.tsx:21]: For the scenario, is it like firing an event to `ChevronDownIcon` and see if there's disclosure content that is visible?
image

**Comment by syauqy** [comment]
Hi Mas @zainfathoni, thank you for the reviews 🙇‍♂️ 🙇‍♂️ 

I've added new commits. It should covers
- [x] fixing the `config.yml` collection structure and format.
- [x] `config.yml` - fixing names and labels inconsistency
- [x] `config.yml` - move JSON files to `_content/kontak-darurat` 
- [ ] `kontak-darurat.test.tsx` - fix the `no-lone-blocks` issues.
- [ ] `kontak-darurat.text.tsx` - fix the array`.map` issues.
- [ ] move all type definition files to `lib/content` directory.
- [ ] `kontak-darurat.text.tsx` - add mock data using `@jackfranklin/test-data-bot` builder.
- [ ] `kontak-darurat.text.tsx` - add test case for `ChevronDownIcon` interaction

File: __tests__/pages/kontak-darurat.test.tsx [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+  });
+
+  it("renders the vaccine disclosure correctly", () => {
+    render(<VaccineSection vaccine_section={vaccine_section} />);
+    const disclosure_title = screen.getByText(/Mau Vaksin COVID-19?/i);
+    expect(disclosure_title).toBeVisible();
+
+    expect(screen.queryByText(vaccine.title)).toBeNull();
+    fireEvent.click(screen.getByTestId("chevron-down-icon"));
+    expect(screen.queryByText(vaccine.title)).toBeVisible();
```
Comment (zainfathoni) [inline __tests__/pages/kontak-darurat.test.tsx:109]:
```suggestion
    expect(screen.queryByText(vaccine.title)).not.toBeInTheDocument();
    userEvent.click(screen.getByTestId("chevron-down-icon"));
    expect(screen.getByText(vaccine.title)).toBeVisible();
```

Please use the`.not.toBeInTheDocument()` assertion from `jest-dom` to make the error message more obvious when the test is failing.
Please use [userEvent instead of fireEvent](https://testing-library.com/docs/guide-events/#interactions-vs-events).
When you're querying for visibility, please use `getByText` instead, because it will throw an error if the element can't be found and it's what we're expecting from assertion of elements that's supposed to be visible.

File: components/layout/stacked-link-disclosure.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (zainfathoni) [reply components/layout/stacked-link-disclosure.tsx:21]: Yes, something like that, but I just added a comment to make the assertions and the action triggers better. 😉

**Review by zainfathoni — CHANGES_REQUESTED** [review]
Great, thanks for making the changes!  🙏 

Almost all of them are addressed properly, I just have a minor comment on your assertions and how you trigger user interactions in the tests. 😁

File: components/layout/stacked-link-disclosure.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (resir014) [reply components/layout/stacked-link-disclosure.tsx:21]: @syauqy Also replace this logic with `clsx`, as with the others.

[the remark this answers is not in the document]
Comment (syauqy) [reply components/layout/stacked-link-disclosure.tsx]: Got it, thank you for the suggestions! mas @zainfathoni mas @resir014  🙇

**Review by zainfathoni — APPROVED** [review]
LGTM 💯 

Thanks a lot @syauqy! 🙏

**Comment by mazipan** [comment]
@resir014 you still block this PR, do you still have any comments need to be done?
